enforcer_cached.go

```diff
@@ -59,44 +64,149 @@ func (e *CachedEnforcer) Enforce(rvals ...interface{}) (bool, error) {
 		return e.Enforcer.Enforce(rvals...)
 	}
 
-	var key strings.Builder
-	for _, rval := range rvals {
-		if val, ok := rval.(string); ok {
-			key.WriteString(val)
-			key.WriteString("$$")
-		} else {
-			return e.Enforcer.Enforce(rvals...)
-		}
+	key,ok := e.getKey(rvals...)
+	if !ok {
+		return e.Enforcer.Enforce(rvals...)
 	}
 
-	if res, ok := e.getCachedResult(key.String()); ok {
+	if res, err := e.getCachedResult(key); err == nil {
 		return res, nil
+	}else if err != ErrNoSuchKey {
```

Comment:
if `err != ErrNoSuchKey`, shall we return `false, err` directly?

Reply:
That's what I did here.

Reply:
👍